e2e-cypress/files/support/login-functions.ts

```diff
@@ -78,17 +78,27 @@ export function addLoginToAAD() {
 export function addSessionLoginWithMFA() {
   Cypress.Commands.add('sessionLoginWithMFA', (username: string, password: string) => {
     cy.session('login', () => cy.loginToAADWithMFA(username, password), {
-      validate: () => cy.getAllLocalStorage().should(validateLocalStorage),
-      cacheAcrossSpecs: true
+      validate: () => new Promise((resolve, reject) => {
+        try {
+          cy.getAllLocalStorage()
+            .should(validateLocalStorage)
+            .then(() => {
+              resolve()
+            })
+        } catch (e) {
+          reject(e)
+        }
+      }),
+      cacheAcrossSpecs: true,
     })
   })
 }
 
 export function addLoginToAADWithMFA() {
   Cypress.Commands.add('loginToAADWithMFA', (username: string, password: string) => {
     loginViaAAD(username, password);
-    cy.getTOTP().then((otp1) => {
-      const objOTP = { otp: otp1 }
+    cy.getTOTP().then((otp) => {
+      const objOTP = { otp }
       cy.origin('https://login.microsoftonline.com/', { args: objOTP }, ({ otp }) => {
```

Comment:
I would not use the additional const here and do `{ args: { otp }}` directly:
```suggestion
      cy.origin('https://login.microsoftonline.com/', { args: { otp } }, ({ otp }) => {
```